### PR TITLE
Cartpage

### PR DESCRIPTION
**Pull request by Shailseen**
### Cart Page 
**Features are added**
- Add cart to items
- Increase/Decrease cart items 
- Remove cart Items
- Cart notification 

**Code to be review**

- Using cart-context to using Cart-Array anywhere
- all axios call in cart-context

https://user-images.githubusercontent.com/90495133/161354735-4ae6ed68-730c-4bdf-82b6-693c2df7e4d9.mp4

### CODE REVIEW

File: src/components/product-card/ProductCard.jsx [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+const ProductCard = ({ item }) => {
+  const { addToCartHandler, cartList } = useCart();
+  let navigate =useNavigate();
+  const goToCart = () => {
+    navigate("/cart");
+  };
   return (
-    <div style={item.includeStock===false ? {opacity:0.3} : {opacity: 1}} className="card-vertical box-shadow" key={item.id}>
+    <div
+      style={item.includeStock === false ? { opacity: 0.3 } : { opacity: 1 }}
```
Comment (hrishikeshmane) [inline src/components/product-card/ProductCard.jsx:13]: avoid inline styling

File: src/components/product-card/ProductCard.jsx [hunk 2]
```diff
@@ -15,19 +28,62 @@ const ProductCard = ({item}) => {
...
+            </button>
+          ) : (
+            <button
+              className="button-style-none solid-button w-100"
+              onClick={goToCart}
+            >
+              Go to Cart
+            </button>
+          )
+        )} */}
```
Comment (hrishikeshmane) [inline src/components/product-card/ProductCard.jsx]: Do not push commented code.

File: src/components/product-card/ProductCard.jsx [hunk 3]
```diff
@@ -15,19 +28,62 @@ const ProductCard = ({item}) => {
...
 
 export { ProductCard };
+// {cartList.map((it) => it._id === item._id ? <><i className="fas fa-cart-plus"></i>Add to cart</> : <></>)}
+
+// <button
+//   className="button-style-none solid-button w-100"
+//   onClick={() => addToCartHandler(item)}
+// >
+//   <i className="fas fa-cart-plus"></i> Add to Cart
+// </button>
```
Comment (hrishikeshmane) [inline src/components/product-card/ProductCard.jsx]: same here

**Review by hrishikeshmane — COMMENTED** [review]
Looks good, apart form the comments above. Make sure to format all files with prettier before pushing.

File: src/components/product-card/ProductCard.jsx (hunk 2 above)
Comment (Shailseen) [reply src/components/product-card/ProductCard.jsx]: okay.

File: src/components/product-card/ProductCard.jsx (hunk 3 above)
Comment (Shailseen) [reply src/components/product-card/ProductCard.jsx]: done

File: src/components/cart-with-items/cartWithItems.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+export const CartItems = () => {
+  const { cartList } = useCart();
+  return (
+    <div className="body-container-grid">
+      <div>
+        {cartList.map((item) => {
+          return <ProductCartCard item={item} />;
+        })}
+      </div>
+      <CartPriceCard />
```
Comment (vishalg8454) [inline src/components/cart-with-items/cartWithItems.jsx]: `key` prop is missing

File: src/components/navbar/Navbar.jsx [hunk 5]
```diff
@@ -34,10 +36,10 @@ const Navbar = () => {
...
 
           <div className="icon-label-container">
-            <a>
+            <Link to="/cart">
               <i className="far fa-shopping-bag"></i>
-            </a>
-            <div className="badge badge-icon-card">6</div>
+            </Link>
+            {cartLength === 0 ? <></> : <div className="badge badge-icon-card">{cartLength}</div>}
           </div>
```
Comment (vishalg8454) [inline src/components/navbar/Navbar.jsx]: && can be used instead of ? :

File: src/components/product-card/ProductCard.jsx (hunk 1 above)
Comment (Shailseen) [reply src/components/product-card/ProductCard.jsx:13]: I dont have approach how to do in css to select styling. i will do it later.

**Review by vishalg8454 — APPROVED** [review]
All good, just couple of suggestions. Can merge

File: src/components/navbar/Navbar.jsx (hunk 5 above)
Comment (Shailseen) [reply src/components/navbar/Navbar.jsx]: thanks for point it.

File: src/components/cart-with-items/cartWithItems.jsx (hunk 4 above)
Comment (Shailseen) [reply src/components/cart-with-items/cartWithItems.jsx]: okay do the changes.

**Pull request merged**